src/lists.rs

```diff
@@ -41,11 +47,38 @@ pub struct ListFormatting<'a> {
     pub h_width: usize,
     // Available width if we layout vertically
     pub v_width: usize,
+    // Non-expressions, e.g. items, will have a new line at the end of the list.
+    // Important for comment styles.
+    pub is_expression: bool
```

Comment:
Could this be named to be more about what you're doing rather than why? ends_with_newline or ends_block or something? (also needs a trailing comma)